src/Core/Hook/Provider/GridDefinitionHookByServiceIdsProvider.php

```diff
@@ -0,0 +1,152 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Hook\Provider;
+
+use Generator;
+use PrestaShop\PrestaShop\Core\Grid\Definition\Factory\GridDefinitionFactoryInterface;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides hooks list by calling service ids from the container.
+ */
+final class GridDefinitionHookByServiceIdsProvider implements HookByServiceIdsProviderInterface
+{
+    const HOOK_STARTS_WITH = 'action';
+    const GRID_DEFINITION_HOOK_ENDS_WITH = 'GridDefinitionModifier';
+    const GRID_QUERY_BUILDER_HOOK_ENDS_WITH = 'GridQueryBuilderModifier';
+    const GRID_DATA_HOOK_ENDS_WITH = 'GridDataModifier';
+    const GRID_FILTER_FORM_ENDS_WITH = 'GridFilterFormModifier';
+    const GRID_PRESENTER_ENDS_WITH = 'GridPresenterModifier';
+
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    /**
+     * @param ContainerInterface $container
+     */
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getHookNames(array $gridDefinitionServiceIds)
+    {
+        $gridDefinitionIds = $this->getGridDefinitionIds($gridDefinitionServiceIds);
+
+        $gridDefinitionHookNames = [];
```

Comment:
Do you really need to have 5 different arrays, and then merge them? Is it to "categorize" the hooks togeter?

Reply:
yeah  - I want to keep same hooks in same place rather then mixing them all.

Reply:
Makes sense for me 👍 the code is very readable from this